src/idea.js

```diff
@@ -11,17 +11,16 @@ class Idea {
      localStorage.setItem("storedInformation", JSON.stringify(userIdeas));
 
    }
-
   deleteFromStorage() {
-    getStoredIdeas()
-    var elementId = parseInt(event.target.id, 10);
-    for (var i = 0; i < userIdeas.length; i++){
-      if (userIdeas[i].id === elementId){
-        userIdeas.splice(i, 1)
-        this.saveToStorage()
-      }  
+    // localStorage.clear();
+    // this.saveToStorage();
+    var localStorageIdeas = JSON.parse(localStorage.getItem("storedInformation"));
+    for (var i = 0; i < localStorageIdeas.length; i++){
+      if (localStorageIdeas[i].id === this.id) {
+        localStorageIdeas.splice(i, 1);
```

Comment:
As a general rule, I tend to avoid mutating arrays in favor of creating a new array. There exists an array prototype method (that perhaps was not covered in your Mod 1 array prototype lesson...) that could be of use here.

Reply:
Thanks for this feedback, Jon! I think for what we're accomplishing in this particular method we will leave this mutation in until we learn more about the slice prototype method and how it could be implemented here.